torch_xla/csrc/tensor.h

```diff
@@ -187,14 +191,13 @@ class XLATensor : public torch::lazy::LazyTensor {
 
     xla::OpSharding sharding;
   };
-  using ShardingSpecPtr = std::shared_ptr<ShardingSpec>;
 
   // Annotate the IR value with ShardingSpec.
   void SetShardingSpec(const ShardingSpec& sharding_spec);
   // Clear sharding annotation attached to the IR value and transfer sharded
   // data back to host.
   void ClearShardingSpec();
-  ShardingSpecPtr sharding_spec() const;
+  ShardingSpecPtr sharding_spec() const { return data()->sharding; }
```

Comment:
add an assert here to make sure `IR`'s sharding spec(if there is one) matches XLATensor's sharding spec. 

ideally we also need to force IR's shardingspec immutable, I think we will sort of do that in Steven's follow up pr where shardingspec becomes an constructor argument.

Reply:
THanks, done!